### PR TITLE
Handle errors gracefully instead of failing in GC Benchmarking Infra

### PR DESCRIPTION
**Pull request by ivdiazsa**
Whenever any benchmark test fails in GC Benchmarking Infra, the entire run is automatically finished at that point. While this is not so bad when running individual tests, it can certainly hamper productivity when running big suites.

This PR introduces a new mechanism, which captures any errors occurred while running benchmarks, and allows tests to continue running. At the end, it displays the user a summary of the problems encountered, if any. It is worth mentioning it comes into play on both previously mentioned scenarios:
- Individual Tests with `py . run`
- Suite Tests with `py . suite-run`

Here are some output examples with an artificially introduced exception:

**Individual Test**

``` text
*WARNING*: Test 'C:\Git\performance\src\benchmarks\gc\bench\suite\normal_workstation.yaml' encountered errors.

*** Here is a summary of the problems found: ***

=== Benchmark: '0gb' ===

Error Message: Pause Exception.

Stack Trace:
  File ".\src\exec\run_tests.py", line 182, in _run_all_benchmarks
    t.out,
  File ".\src\exec\run_single_test.py", line 110, in run_single_test
    partial_test_status = _do_run_single_test(built, t, out)
  File ".\src\exec\run_single_test.py", line 186, in _do_run_single_test
    return _run_single_test_windows_perfview(built, t, out)
  File ".\src\exec\run_single_test.py", line 369, in _run_single_test_windows_perfview
    assert False, "Pause Exception."
```

**Suite Test**

``` text
*WARNING*: One or more tests in the suite encountered errors.

*** Here is a summary of the problems found: ***

======= C:\Git\performance\src\benchmarks\gc\bench\suite\normal_workstation.yaml =======


=== Benchmark: '0gb' ===

Error Message: Pause Exception.

Stack Trace:
  File ".\src\exec\run_tests.py", line 182, in _run_all_benchmarks
    t.out,
  File ".\src\exec\run_single_test.py", line 110, in run_single_test
    partial_test_status = _do_run_single_test(built, t, out)
  File ".\src\exec\run_single_test.py", line 186, in _do_run_single_test
    return _run_single_test_windows_perfview(built, t, out)
  File ".\src\exec\run_single_test.py", line 369, in _run_single_test_windows_perfview
    assert False, "Pause Exception."


======= C:\Git\performance\src\benchmarks\gc\bench\suite\normal_server.yaml =======


=== Benchmark: '0gb' ===

Error Message: Pause Exception.

Stack Trace:
  File ".\src\exec\run_tests.py", line 182, in _run_all_benchmarks
    t.out,
  File ".\src\exec\run_single_test.py", line 110, in run_single_test
    partial_test_status = _do_run_single_test(built, t, out)
  File ".\src\exec\run_single_test.py", line 186, in _do_run_single_test
    return _run_single_test_windows_perfview(built, t, out)
  File ".\src\exec\run_single_test.py", line 369, in _run_single_test_windows_perfview
    assert False, "Pause Exception."
```

Additionally, I added a nice message that is displayed when a `suite-run` finishes successfully!

@Maoni0 @VSadov @cshung

### CODE REVIEW

File: src/benchmarks/gc/src/exec/run_tests.py [hunk 1]
```diff
@@ -119,23 +160,34 @@ def _run_all_benchmarks(
...
+                        test_exe=_get_path(built, t.bench_file.paths, t.benchmark.get_executable),
+                        options=t.bench_file.options,
+                        default_env=default_env,
+                    ),
+                    t.out,
+                )
+        except Exception:
+            _, exception_message, exception_trace = exc_info()
+            run_errors.append(
+                BenchmarkErrorInfo(t.benchmark_name,
```
Comment (Maoni0) [inline src/benchmarks/gc/src/exec/run_tests.py:186]: this gives the benchmark name but you could have different configs the same benchmark gets run with. can we also give the configs? the purpose of providing such errors is to help whoever investigates the error to construct the failure scenario and rerun it.

Comment (ivdiazsa) [reply src/benchmarks/gc/src/exec/run_tests.py:186]: I'm so used to running one configuration I forgot about that functionality. Will add it to the summary messages as well.

Comment (Maoni0) [reply src/benchmarks/gc/src/exec/run_tests.py:186]: the coreclr build is also parameter that can vary, from the doc:
"The test runner uses all combinations of coreclrs ⨯ configs ⨯ benchmarks."

**Comment by ivdiazsa** [comment]
Updated the summary shown after a test or suite finishes running. Now, it will display the information from each Core_Root tested, as well as each configuration run. Here are some examples:

- **Individual Test** (`py . run`)

``` text
======= *WARNING*: Test 'path/to/bench/suite/normal_workstation.yaml' encountered errors. =======

*** Here is a summary of the problems found: ***

===== Core 'a' =====

=== Configuration 'smaller' ===

- Benchmark: '0gb' -
Iteration: 0
Error Message: Pause Exception.

Stack Trace:
  File ".\src\exec\run_tests.py", line 181, in _run_all_benchmarks
    t.out,
  File ".\src\exec\run_single_test.py", line 110, in run_single_test
    partial_test_status = _do_run_single_test(built, t, out)
  File ".\src\exec\run_single_test.py", line 181, in _do_run_single_test
    assert False, "Pause Exception."

<repeat with other benchmarks that failed>

=== Configuration 'bigger' ===

- Benchmark: '0gb' -
Iteration: 0
Error Message: Pause Exception.

Stack Trace:
  File ".\src\exec\run_tests.py", line 181, in _run_all_benchmarks
    t.out,
  File ".\src\exec\run_single_test.py", line 110, in run_single_test
    partial_test_status = _do_run_single_test(built, t, out)
  File ".\src\exec\run_single_test.py", line 181, in _do_run_single_test
    assert False, "Pause Exception."

<repeat with other benchmarks that failed>


===== Core 'b' =====

<show all configurations and benchmarks that failed with this core_root>
```

- **Suite Test** (`py . suite-run`)

``` text
*WARNING*: One or more tests in the suite encountered errors.

*** Here is a summary of the problems found: ***

======= path/to/bench/suite/normal_workstation.yaml =======

<show all errors that occurred when running this test with the format described above>


======= path/to/bench/suite/normal_server.yaml =======

<show all errors that occurred when running this test with the format described above>

<repeat with any other tests included in the suite.yaml file>
```

**Comment by ivdiazsa** [comment]
What do you think about this new summary format? @Maoni0

**Comment by ivdiazsa** [comment]
@billwert @brianrob This PR is ready for merging.

**Pull request merged**